pkg/broker/handler/processors/deliver/processor.go

```diff
@@ -176,6 +177,13 @@ func (p *Processor) deliver(ctx context.Context, target *config.Target, broker *
 		return fmt.Errorf("event delivery failed: HTTP status code %d", resp.StatusCode)
 	}
 
+	// Pre-check the reply response header, if it's not in structured mode/batched mode or binary mode,
+	// then it's not a CloudEvent, we treat the delivery as successful and ignore the response.
+	// Otherwise, we proceed with the malformed event check.
+	if !strings.HasPrefix(resp.Header.Get("Content-Type"), "application/cloudevents") && (resp.Header.Get("ce-specversion") == "") {
```

Comment:
I don't think this is accurate. I believe in binary mode the `Content-Type` can be just about anything, see https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#311-http-content-type.

In fact the first example in https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md#314-examples is:

```
POST /someresource HTTP/1.1
Host: webhook.example.com
ce-specversion: 1.0
ce-type: com.example.someevent
ce-time: 2018-04-05T03:56:24Z
ce-id: 1234-1234-1234
ce-source: /mycontext/subcontext
    .... further attributes ...
Content-Type: application/json; charset=utf-8
Content-Length: nnnn

{
    ... application data ...
}
```

Note that `Content-Type` is `application/json; charset=utf-8`, which is _not_ prefixed with `application/cloudevents`.

Reply:
I am completely wrong. Please ignore.